src/main/parser.js

```diff
@@ -42,14 +42,19 @@ function resolveParser(opts, parsers) {
     if (parsers.hasOwnProperty(opts.parser)) {
       return parsers[opts.parser];
     }
-    try {
-      return {
-        parse: eval("require")(path.resolve(process.cwd(), opts.parser)),
-        astFormat: "estree",
-        locStart,
-        locEnd
-      };
-    } catch (err) {
+    if (process.env.PRETTIER_TARGET !== "universal") {
+      try {
+        return {
+          parse: eval("require")(path.resolve(process.cwd(), opts.parser)),
+          astFormat: "estree",
+          locStart,
+          locEnd
+        };
+      } catch (err) {
+        /* istanbul ignore next */
+        throw new ConfigError(`Couldn't resolve parser "${opts.parser}"`);
+      }
+    } else {
       /* istanbul ignore next */
       throw new ConfigError(`Couldn't resolve parser "${opts.parser}"`);
```

Comment:
Something like _"Resolving parser is not supported in non-Node environment"_ should be better I guess.